src/qibolab/result.py

```diff
@@ -5,97 +5,200 @@
 import numpy as np
 import numpy.typing as npt
 
-ExecRes = np.dtype([("i", np.float64), ("q", np.float64)])
+iq_typing = np.dtype([("i", np.float64), ("q", np.float64)])
 
 
-@dataclass
-class ExecutionResults:
-    """Data structure to deal with the output of :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`"""
-
-    array: npt.NDArray[ExecRes]
-    shots: Optional[npt.NDArray[np.uint32]] = None
+class IQNotEqualLenght(Exception):
+    def __init__(self, message="is_ and qs_ must have the same size"):
+        super().__init__(message)
 
-    @classmethod
-    def from_components(cls, is_, qs_, shots=None):
-        ar = np.empty(is_.shape, dtype=ExecRes)
-        ar["i"] = is_
-        ar["q"] = qs_
-        ar = np.rec.array(ar)
-        return cls(ar, shots)
 
-    @property
-    def i(self):
-        return self.array.i
-
-    @property
-    def q(self):
-        return self.array.q
-
-    def __add__(self, data):
-        assert len(data.i.shape) == len(data.q.shape)
-        if data.shots is not None:
-            assert len(data.i.shape) == len(data.shots.shape)
-        # concatenate on first dimension; if a scalar is passed, just append it
-        axis = 0 if len(data.i.shape) > 0 else None
-        i = np.append(self.i, data.i, axis=axis)
-        q = np.append(self.q, data.q, axis=axis)
-        if data.shots is not None:
-            shots = np.append(self.shots, data.shots, axis=axis)
-        else:
-            shots = None
-
-        new_execution_results = self.__class__.from_components(i, q, shots)
-
-        return new_execution_results
+@dataclass
+class IQResults:
```

Comment:
Maybe it would make sense to name this
```suggestion
class IntegratedResults:
```
to be closer to `AcquisitionType.INTEGRATION` that results to this type of result. Also IQ appear in other places (eg. the raw waveforms), so it is better to explicitly say that here it is integrated.

If you change it, you should also change `AveragedIQResults` to `AveragedIntegratedResults`.

Reply:
Yeah I should do that, I named like that because I had waveforms and integrated as one there.